sync.js

```diff
@@ -430,7 +434,10 @@ class Sync extends events.EventEmitter {
 
       function onClose (err) {
         disconnected = true
-        if (peer) peer.connected = false
+        if (peer) {
+          peer.connected = false
+          if (peer.started) self.osm.core.resume()
```

Comment:
Was there a reason why osm.core.resume() was happening in the end of pump rather than here onClose?

Reply:
Yes. `onClose` is called even if the connection is lost before sync starts (see the event listeners on `connection` above). If it's called there and `resume`s, it could unbalance the pause/resume parity.